enforcer.go

```diff
@@ -416,6 +425,33 @@ func (e *Enforcer) enforce(matcher string, rvals ...interface{}) (ok bool, err e
 
 			parameters.pVals = pvals
 
+			if hasEval {
+				ruleNames := util.GetEvalValue(expString)
+				if err != nil {
+					return false, err
+				}
+
+				var expWithRule = expString
+				for i, ruleName := range ruleNames {
+					if i%2 == 0 {
```

Comment:
@hsluoyz I'm not sure it's correct here.

Reply:
This is because the first value will be the matched string itself, and now I moved this code to `GetEvalValue ()` and added some unit tests